rules/no-new-statics.js

```diff
@@ -0,0 +1,20 @@
+'use strict'
+
+const PROMISE_STATICS = require('./lib/promise-statics')
+
+module.exports = function (context) {
+  return {
+    NewExpression (node) {
+      if (
+        node.callee.type === 'MemberExpression' &&
+        node.callee.object.name === 'Promise' &&
+        PROMISE_STATICS.indexOf(node.callee.property.name) > -1
+      ) {
+        context.report({
+          node,
+          message: 'Avoid calling new on a Promise static method'
```

Comment:
This is a generic message right now that helps get the point across (I think). Improvement suggestions welcome.